spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamDefinition.java

```diff
@@ -153,6 +149,28 @@ public void remove() {
 			throw new UnsupportedOperationException();
 		}
 
+		/**
+		 * Return whether or not this iterator is past the very first module to deploy.
+		 */
+		public boolean hasMoreDownstream() {
+			return i < modules.size() - 2;
```

Comment:
I don't understand how this could work. If a stream has 2 modules e.g. "time | log" and `next()` has not yet been called, on this iterator, `i` would have a value of 1. In that case this method would return `false` which is *correct* since nothing has been passed yet. Then, however, if `next()` is called, `i` would have a value of 0, and this would still return `false`.